integration-tests/src/checks/poi_divergence_local.rs

```diff
@@ -0,0 +1,89 @@
+use crate::utils::RadioRuntimeConfig;
+use colored::Colorize;
+use poi_radio::{LocalAttestationsMap, MessagesArc, RemoteAttestationsMap};
+use tracing::{debug, info};
+
+use crate::setup::test_radio::run_test_radio;
+
+fn post_comparison_handler(_messages: MessagesArc, _block: u64, _subgraph: &str, _prev_len: usize) {
+}
+
+fn test_attestation_handler(
+    block: u64,
+    remote: &RemoteAttestationsMap,
+    local: &LocalAttestationsMap,
+) {
+    debug!("Something from test_attestation_handler");
+    debug!("Inputs {:?}{:?}{:?}", block, remote, local);
+
+    if let Some((ipfs_hash, blocks)) = local.iter().next() {
+        if let Some(local_attestation) = blocks.get(&block) {
+            if let Some(remote_blocks) = remote.get(ipfs_hash) {
+                if let Some(remote_attestations) = remote_blocks.get(&block) {
+                    let mut remote_attestations = remote_attestations.clone();
+                    remote_attestations.sort_by(|a, b| a.stake_weight.cmp(&b.stake_weight));
+
+                    debug!(
+                        "Remote attestations on block {}: {:#?}",
+                        block, remote_attestations
+                    );
+                    info!(
+                        "Number of nPOIs submitted for block {}: {:#?}",
+                        block,
+                        remote_attestations.len()
+                    );
+                    if remote_attestations.len() > 1 {
+                        info!("Sorted attestations: {:#?}", remote_attestations);
+                    }
+
+                    let most_attested_npoi = &remote_attestations.last().unwrap().npoi;
+                    info!("Most attested npoi: {:#?}", most_attested_npoi);
+                    info!("Local npoi: {:#?}", &local_attestation.npoi);
+
+                    // This check assumes there are less basic than divergent instances running. This is what remote attestations looks like currently with no divergence (example):
```

Comment:
do we want this comment? could the attestation details be tested with expected stake sum and sender addr?

Reply:
ideally they would, but since the network can lag behind the messages we get are not deterministic, that's why I left the check a bit loose, but you're right the comment just adds noise, will remove

Reply:
✅